### PR TITLE
Seabeta 200 email infrastructure

### PR DESCRIPTION
**Pull request by CodrinCotarlanSoftwire**
Comments about file structure / naming are welcome :)

### CODE REVIEW

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

**Comment by CodrinCotarlanSoftwire** [comment]
Moved entire email templates in he config file. Now, the entire structure of govuk emails is bound in the 'Startup' section and easily used as a service in the NotifyApi file.

File: SeaPublicWebsite/Helpers/Global.cs [hunk 1]
```diff
@@ -13,5 +13,7 @@ public static class Global
         public static string BasicAuthPassword => Config.GetAppSetting("BasicAuthPassword");
         public static string EpcAuthUsername => Config.GetAppSetting("EpcAuthUsername");
         public static string EpcAuthPassword => Config.GetAppSetting("EpcAuthPassword");
+        public static string GovUkNotifyApiKey => Config.GetAppSetting("GovUkNotifyApiKey");
+        public static string GovUkNotifyApiTestKey => Config.GetAppSetting("GovUkNotifyApiTestKey");
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/Helpers/Global.cs:17]: This isn't used? If that's right, please remove it

File: SeaPublicWebsite/appsettings.json [hunk 2]
```diff
@@ -6,5 +6,17 @@
             "Microsoft.Hosting.Lifetime": "Information"
         }
     },
-    "AllowedHosts": "*"
+    "AllowedHosts": "*",
+    
+    "GovUkNotify": {
+        "Test": "test",
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/appsettings.json:12]: Please remove :)

File: SeaPublicWebsite/appsettings.json [hunk 3]
```diff
@@ -6,5 +6,17 @@
             "Microsoft.Hosting.Lifetime": "Information"
         }
     },
-    "AllowedHosts": "*"
+    "AllowedHosts": "*",
+    
+    "GovUkNotify": {
+        "Test": "test",
+        "ApplicationReferenceNumber": {
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/appsettings.json:13]: Can we make it clearer what these configuration values are for?
Maybe put these in a `"TemplateDetails": {` section? Or put `Template` at the end of each name (so `ApplicationReferenceNumberTemplate`.
Inside each section please make it clear that the other values are the names of placeholders. Maybe just start all the names with `placeholder` or similar.

File: SeaPublicWebsite/ExternalServices/EmailSending/GovUkNotifyConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿namespace SeaPublicWebsite.ExternalServices.EmailSending
+{
+    public class GovUkNotifyConfiguration
+    {
+        public const string Name = "GovUkNotify";
+        public string Test { get; set; }
+        public ApplicationReferenceNumberConfiguration ApplicationReferenceNumber { get; set; }
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/ExternalServices/EmailSending/GovUkNotifyConfiguration.cs:7]: It seems a bit unnecessary to have three separate files for this config object. As the classes are only for storing information I think it's fine to make `ApplicationReferenceNumberConfiguration` and `RequestDocumentConfiguration` inner classes in this file.

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

File: SeaPublicWebsite/Helpers/Global.cs (hunk 1 above)
Comment (CodrinCotarlanSoftwire) [reply SeaPublicWebsite/Helpers/Global.cs:17]: The Test key is not used, I'll remove that.
The regular Api key is, the api needs it

File: SeaPublicWebsite/appsettings.json (hunk 3 above)
Comment (CodrinCotarlanSoftwire) [reply SeaPublicWebsite/appsettings.json:13]: +1 For adding 'Template' at the end of the name and for the placeholder suggestion :)
If only there was a way to make the names smaller.. oh well, clarity beats bad naming!

File: SeaPublicWebsite/ExternalServices/EmailSending/GovUkNotifyConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿namespace SeaPublicWebsite.ExternalServices.EmailSending
+{
+    public class GovUkNotifyConfiguration
+    {
+        public const string Name = "GovUkNotify";
+        public ApplicationReferenceNumberConfiguration ApplicationReferenceNumberTemplate { get; set; }
+        public RequestDocumentConfiguration RequestDocumentTemplate { get; set; }
+    }
+    public class ApplicationReferenceNumberConfiguration
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/ExternalServices/EmailSending/GovUkNotifyConfiguration.cs:9]: blank line please :)

**Review by DanCorderSoftwire — APPROVED** [review]

**Pull request merged**